### ISSUE TITLE
Getting wrong value in "renderMonthElement"

### ISSUE THREAD

**Issue by nitish-arora** [airbnb/react-dates#2165]
**react-dates version**
react-dates@^21.8.0

**Describe the bug**
Getting wrong value for month in renderMonthElement 

  const [dates, setDates] = useState({ startDate: moment(), endDate: moment().endOf("week") });

```
    <DayPickerRangeController
      startDate={dates.startDate}
      endDate={dates.endDate}
      focusedInput={focusedInput}
      onDatesChange={onDatesChange}
      onFocusChange={onFocusChange}
      startDateOffset={_startDayOffsetFn}
      endDateOffset={_endDayOffsetFn}
      enableOutsideDays
      numberOfMonths={1}
      hideKeyboardShortcutsPanel
      daySize={35}
      minDate={moment()}
      maxDate={moment().add(3, "months")}
      isOutsideRange={_validDatesFn}
      renderMonthElement={({month, onMonthSelect}) => {
        console.log("s", month.month());
        console.log("y", month.year())
        console.log("month", month);
        return month.format("MMMM YYYY");
      }}
    />
```

**Desktop (please complete the following information):**
 - OS:  Windows
 - Browser: Chrome
 - Version: 96.0.4664.45

**Comment by ljharb** [comment]
What value are you getting, and what do you expect?

**Comment by nitish-arora** [comment]
I am getting 3 different values, same date for
- previous month 
- current month
- next month

**Comment by ljharb** [comment]
You're console logging 3 things. It's very unclear to me whether these three things are logging correctly or not, what values you expect here, and what you're getting instead.

Could you perhaps make a repro test case, using a repo or a codesandbox?

**Comment by nitish-arora** [comment]
Yes i know but for month.month() i am getting 10,11, 0 one after another.
I don’t know why this is happening.

**Comment by ljharb** [comment]
Because it's rendering November, and then December, and then January - `.month()` returns an index of an array of month names, so `0` is January and `11` is December.

**Comment by nitish-arora** [comment]
Shouldn’t i get a single value. Why three? And why November & January? I am
using current date as start date.

**Comment by ljharb** [comment]
You get one value for each month that's rendered. Your max date explicitly says "3 months after the min date" - so your range covers 3 months, and either way the datepicker renders the previous month, the current month, and the next month.

**Comment by nitish-arora** [comment]
Oh is it, can i know the complete documentation where i can get to know
about all the props?

**Comment by ljharb** [comment]
It's all in or linked from the readme.

**Comment by nitish-arora** [comment]
Readme is giving the details about props like component will accept these props but i am not able to get to know what exactly these props are for like renderMonthElement

**Comment by nitish-arora** [comment]
![image](https://user-images.githubusercontent.com/45289433/144993916-4f989c4a-20db-440f-9349-a8acd09812fb.png)


I am getting dates of three months with next year, this is the problem I am facing.

**Comment by nitish-arora** [comment]
I actually want to update the month and year based on startDate and i can change startDate from different action aside from calendar also. So whenever i click on that button it should update the month and year in calendar based on that startDate

**Comment by nitish-arora** [comment]
Is there some method or api by which I can do that.
I was trying this 
const ref = useRef();

useEffect(() => {
   ref.current.onMonthSelect(ref.current.month, startDate.format("M"))
},[startDate])


renderMonthElement = {({month, onMonthSelect, onYearSelect}) => {
 ref.current = {month, onMonthSelect, onYearSelect};
}

}


but its not working, it is giving the response as in the image I uploaded above

**Comment by akf-wtag** [comment]
@nitish-arora have you solved the problem? I'am facing the exact same issue and the docs don't say much about these props.
